src/UserAdmin/src/Smart.FA.Catalog.Web/Pages/404.cshtml.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace Smart.FA.Catalog.Web.Pages;
+
+public class NotFoundPageModel : PageModel
+{
+    public const string ErrorTitleKey = nameof(ErrorTitle);
+
+    public const string ErrorMessageKey = nameof(ErrorMessageKey);
+
+    public string ErrorTitle { get; set; } = null!;
+
+    public string ErrorMessage { get; set; } = null!;
+
+    public PageResult OnGet()
+    {
+        SetTitle();
+        SetMessage();
+
+        return Page();
+    }
+
+    private void SetTitle()
+    {
+        if (TempData.ContainsKey(ErrorTitleKey))
+        {
+            ErrorTitle = TempData[ErrorTitle]!.ToString()!;
+        }
+        else if (HttpContext.Items.ContainsKey(ErrorTitleKey))
+        {
+            ErrorTitle = HttpContext.Items[ErrorTitleKey]!.ToString()!;
+        }
+        else
+        {
+            ErrorTitle = CatalogResources.Sorry;
+        }
```

Comment:
We could use return statements instead of if-else clauses
```suggestion
        if (TempData.ContainsKey(ErrorTitleKey))
        {
            ErrorTitle = TempData[ErrorTitle]!.ToString()!;
            return;
        }
        
        if (HttpContext.Items.ContainsKey(ErrorTitleKey))
        {
            ErrorTitle = HttpContext.Items[ErrorTitleKey]!.ToString()!;
            return;
        }
       
       ErrorTitle = CatalogResources.Sorry;
        
```

Reply:
I am not a huge fan in those context to use the return statement just to break the code flow.
I think we should aim to use it in most case to return actual value (it's not a absolute rule ofc).
There's many debate on that topic; where people don't agree but I'd rather keep it that way.